feat: Start font preloading early on all platforms

MartinZikmund opened the pull request:
GitHub Issue (If applicable): closes [#1041](https://github.com/unoplatform/uno-private/issues/1041)

<!-- Link to relevant GitHub issue if applicable. All PRs should be associated with an issue (GitHub issue or internal), unless the change is documentation related. -->

## PR Type

What kind of change does this PR introduce?

- Feature

## What is the current behavior?

<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->


## What is the new behavior?

<!-- Please describe the new behavior after your modifications. -->

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below.
     Please note that breaking changes are likely to be rejected -->

## Other information

<!-- Please provide any additional information if necessary -->

Internal Issue (If applicable):
<!-- Link to relevant internal issue if applicable. All PRs should be associated with an issue (GitHub issue or internal) -->

MartinZikmund committed:
feat: Start font preloading early on all platforms

unodevops commented:
🤖 Your WebAssembly Sample App stage site is ready! Visit it here: https://unowasmprstaging.z20.web.core.windows.net/pr-19945/index.html

MartinZikmund committed:
chore: Start loading the symbols font first

unodevops commented:
🤖 Your WebAssembly Sample App stage site is ready! Visit it here: https://unowasmprstaging.z20.web.core.windows.net/pr-19945/index.html

unodevops commented:
🤖 Your Docs stage site is ready! Visit it here: https://unodocsprstaging.z13.web.core.windows.net/pr-19945/index.html

unodevops commented:
⚠️⚠️ The build [161355](https://dev.azure.com/uno-platform//Uno%20Platform/_build/results?buildId=161355) has failed on Uno.UI - CI.

MartinZikmund committed:
fix: Adjust namespacing for FontFamilyHelper

MartinZikmund committed:
refactor: Move public FontFamilyHelper

unodevops commented:
🤖 Your WebAssembly Sample App stage site is ready! Visit it here: https://unowasmprstaging.z20.web.core.windows.net/pr-19945/index.html

unodevops commented:
🤖 Your Docs stage site is ready! Visit it here: https://unodocsprstaging.z13.web.core.windows.net/pr-19945/index.html

MartinZikmund committed:
chore: Adjust namespace

unodevops commented:
🤖 Your WebAssembly Sample App stage site is ready! Visit it here: https://unowasmprstaging.z20.web.core.windows.net/pr-19945/index.html

unodevops commented:
🤖 Your Docs stage site is ready! Visit it here: https://unodocsprstaging.z13.web.core.windows.net/pr-19945/index.html

unodevops commented:
⚠️⚠️ The build [161613](https://dev.azure.com/uno-platform//Uno%20Platform/_build/results?buildId=161613) has failed on Uno.UI - CI.

MartinZikmund commented:
@mergifyio backport release/stable/6.0

unodevops commented:
🤖 Your WebAssembly Sample App stage site is ready! Visit it here: https://unowasmprstaging.z20.web.core.windows.net/pr-19945/index.html

unodevops commented:
🤖 Your Docs stage site is ready! Visit it here: https://unodocsprstaging.z13.web.core.windows.net/pr-19945/index.html

MartinZikmund commented:
/azp run

Youssef1313 commented on src/Uno.UI/UI/Xaml/Application.skia.cs:
```diff
@@ -129,6 +133,18 @@ private void InvokeOnLaunched()
+				return;
+			}
+
+			var _ = FontFamilyHelper.PreloadAsync(new FontFamily(FeatureConfiguration.Font.SymbolsFont), FontWeights.Normal, FontStretch.Normal, FontStyle.Normal);
```
```suggestion
			_ = FontFamilyHelper.PreloadAsync(new FontFamily(FeatureConfiguration.Font.SymbolsFont), FontWeights.Normal, FontStretch.Normal, FontStyle.Normal);
```

NOTE: That's a fire-and-forget operation. Usually a thing to avoid as it may lead to unobserved exceptions

unodevops commented:
⚠️⚠️ The build [161751](https://dev.azure.com/uno-platform//Uno%20Platform/_build/results?buildId=161751) has failed on Uno.UI - CI.

MartinZikmund committed:
chore: Apply fix from code review

Co-authored-by: Youssef Victor <youssefvictor00@gmail.com>

unodevops commented:
🤖 Your WebAssembly Sample App stage site is ready! Visit it here: https://unowasmprstaging.z20.web.core.windows.net/pr-19945/index.html

unodevops commented:
🤖 Your Docs stage site is ready! Visit it here: https://unodocsprstaging.z13.web.core.windows.net/pr-19945/index.html

unodevops commented:
⚠️⚠️ The build [161800](https://dev.azure.com/uno-platform//Uno%20Platform/_build/results?buildId=161800) has failed on Uno.UI - CI.

MartinZikmund commented:
/azp run

unodevops commented:
🤖 Your WebAssembly Sample App stage site is ready! Visit it here: https://unowasmprstaging.z20.web.core.windows.net/pr-19945/index.html

unodevops commented:
🤖 Your Docs stage site is ready! Visit it here: https://unodocsprstaging.z13.web.core.windows.net/pr-19945/index.html

unodevops commented:
⚠️⚠️ The build [161828](https://dev.azure.com/uno-platform//Uno%20Platform/_build/results?buildId=161828) has failed on Uno.UI - CI.

unodevops commented:
🤖 Your WebAssembly Sample App stage site is ready! Visit it here: https://unowasmprstaging.z20.web.core.windows.net/pr-19945/index.html

unodevops commented:
🤖 Your Docs stage site is ready! Visit it here: https://unodocsprstaging.z13.web.core.windows.net/pr-19945/index.html

MartinZikmund committed:
fix: Improve font preloading logic

The changes modify the font preloading logic in the `WebAssemblyWindowWrapper` class and the `Application.skia.cs` file. The code now checks if the `FeatureConfiguration.Font.DefaultTextFontFamily` can be converted into a `Uri` before preloading fonts. If successful, it uses the `Uri`; otherwise, it falls back to the original string. This enhancement ensures that the `DefaultTextFontFamily` is a valid `Uri`, improving the robustness of the font preloading process.

MartinZikmund committed:
fix: Use OpenSans in HR app

unodevops commented:
🤖 Your WebAssembly Sample App stage site is ready! Visit it here: https://unowasmprstaging.z20.web.core.windows.net/pr-19945/index.html

MartinZikmund commented:
@Mergifyio backport release/stable/6.0

unodevops commented:
⚠️⚠️ The build [162150](https://dev.azure.com/uno-platform//Uno%20Platform/_build/results?buildId=162150) has failed on Uno.UI - CI.
